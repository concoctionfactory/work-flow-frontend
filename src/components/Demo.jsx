import React, { useState, useEffect } from "react";
import { NavLink as RouterLink } from "react-router-dom";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import { loginUserAPI } from "../actions/users";

import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";
const useStyles = makeStyles((theme) => ({
  root: {
    width: 200,
    height: 200,
  },
  heroContent: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(8, 0, 6),
  },
  heroButtons: {
    marginTop: theme.spacing(4),
  },
}));

function Demo() {
  const classes = useStyles();
  const dispatch = useDispatch();
  const history = useHistory();

  async function handleLogin(e) {
    e.preventDefault();
    await dispatch(loginUserAPI({ username: "alex_ant", password: "12345" }));
  }
  const [text, setText] = useState("");

  let main = (
    <main>
      {/* Hero unit */}
      <div className={classes.heroContent}>
        <Container maxWidth="sm">
          <Typography
            component="h1"
            variant="h2"
            align="center"
            color="textPrimary"
            gutterBottom
          >
            WorkFlow
          </Typography>
          <Typography
            variant="h5"
            align="center"
            color="textSecondary"
            paragraph
          >
            Project management made simple, get your work in a flow. Get started
            by signing up or trying out the demo mode
          </Typography>
          <div className={classes.heroButtons}>
            <Grid container spacing={2} justify="center">
              <Grid item>
                <Button
                  variant="contained"
                  color="primary"
                  component={RouterLink}
                  to="/login"
                >
                  signup/ login
                </Button>
              </Grid>
              <Grid item>
                <Button
                  variant="outlined"
                  color="primary"
                  onClick={handleLogin}
                >
                  demo mode
                </Button>
              </Grid>
            </Grid>
          </div>
        </Container>
      </div>
    </main>
  );

  useEffect(() => {
    const timer = setTimeout(() => setText(main), 200);
    return () => clearTimeout(timer);
  }, []);
  return text;
}

export default Demo;
