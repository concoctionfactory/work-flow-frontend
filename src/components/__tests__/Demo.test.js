import React from "react";
import { render } from "@testing-library/react";
import Demo from "../Demo";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "../../reducers/root";
import { BrowserRouter } from "react-router-dom";
import thunk from "redux-thunk";
const store = createStore(rootReducer, applyMiddleware(thunk));

it("renders without crashing", function () {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Demo />
      </BrowserRouter>
    </Provider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Demo />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows text ", function () {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <Demo />
      </BrowserRouter>
    </Provider>
  );
  expect(component.getByText("WorkFlow")).toBeInTheDocument();
  expect(component.getByText("demo mode")).toBeInTheDocument();
});
