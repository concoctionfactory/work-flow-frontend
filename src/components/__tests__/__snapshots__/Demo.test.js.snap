// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <div
      class="makeStyles-heroContent-5"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <h1
          class="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          WorkFlow
        </h1>
        <p
          class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
        >
          Project management made simple, get your work in a flow. Get started by signing up or trying out the demo mode
        </p>
        <div
          class="makeStyles-heroButtons-6"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                href="/login"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  signup/ login
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  demo mode
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
